### ISSUE TITLE
Inspector.get_filter_parameters and array parameter

### ISSUE THREAD

**Issue by b0g3r** [axnsan12/drf-yasg#317]
I use `django-filter` for filtering via query-strings. My filtersets contains `MultipleChoiceField`, and it parses query-string like `?field=1&field=2` to `field: [1, 2]`. 

But in swagger view from drf-yasg I see only one textfield in parameters for this endpoint. 

After a little research, I found `Inspector.get_filter_parameters`
https://github.com/axnsan12/drf-yasg/blob/583e404ed826416bade3505d1c77dba36b98ba9b/src/drf_yasg/inspectors/query.py#L23-L27
 and  `coreapi_field_to_parameter`.
https://github.com/axnsan12/drf-yasg/blob/583e404ed826416bade3505d1c77dba36b98ba9b/src/drf_yasg/inspectors/query.py#L35-L57
But it looks like it doesn't support array params. Any ideas?

**Comment by b0g3r** [comment]
Also, I found 
https://github.com/carltongibson/django-filter/blob/7471b52a28740213f65f5374cf89f0c693f91c6c/django_filters/rest_framework/backends.py#L134-L139
and
https://github.com/carltongibson/django-filter/blob/master/django_filters/rest_framework/backends.py#L107-L114

Do you have any ideas why django-filter uses only numbers and strings for Parameters?

**Comment by rsichny** [comment]
https://github.com/carltongibson/django-filter/blob/f4981c8644826fc343efcec601bbb91eeb390162/django_filters/filters.py#L498

`status__in = filters.BaseCSVFilter(field_name='status', lookup_expr='in')`

requests look like `?status__in=1,2`

**Comment by axnsan12** [comment]
I think this is a problem with django-filter? It doesn't seem like get_schema_fields from django-filter ever returns an array coreschema.

Something similar was fixed with #257, are you using the latest version of drf-yasg?

It would be great if you could show a complete view/model/serializer/filter-backend combination and the output you would expect vs what is actually generated.

**Comment by rsichny** [comment]
It makes no sense for query params to be anything exept strings, so it's all about processing the strings properly in Filter classes

**Comment by xassbit** [comment]
I'm having the same issue, which I've described in https://stackoverflow.com/q/55818137/6190121.

Indeed the default Django filter backend returns type "string" for all fields except `NumberFilter`. But the filter backend (particularly the method `DjangoFilterBackend.get_coreschema_field`) is easily overridable. However, when adding a clause to return type "array" for multiple choice fields, it fails in `drf-yasg` (more specifically, the method `CoreAPICompatInspector.coreapi_field_to_parameter` fails to convert a core api "array" type to an open api "array" type). I kept editing the `drf-yasg` to see how far I could get with some quick patches, but I stopped when I found that the `Items` class didn't accept `explode` for type array.

In summary, in order to accept serialization of the kind
```yaml
parameters:
- name: type
  in: query
  description: ''
  required: false
  schema:
      type: array
      items:
          type: string
  explode: true
```

two things have to happen:
 - the inspector must identify parameters of the "array" type;
 - the `explode` setting must be accepted for "array" type.

**Comment by marcorichetta** [comment]
> two things have to happen:
> 
>     * the inspector must identify parameters of the "array" type;
> 
>     * the `explode` setting must be accepted for "array" type.

@xassbit Is it possible to add the `explode` argument in the `Parameter` class?

I'm asking this because the OpenAPI v2 doesn't include a reference to explode. The only thing I found is the use of `collectionFormat: multi` in the **Array and Multi-Value Parameters** section. [OpenAPI v2](https://swagger.io/docs/specification/2-0/describing-parameters/#array)

[OpenAPI v3](https://swagger.io/docs/specification/serialization/) includes references to the explode argument, as described by your example. Also it's included in [drf-spectacular](https://github.com/tfranzel/drf-spectacular/blob/8b9d0cafe856d6219cf4f5c4ab04bf5e9ca21268/drf_spectacular/utils.py#L121).

Most important, is this phrase from [v3](https://swagger.io/docs/specification/serialization/#query):
> The default serialization method is style: form and explode: true. This corresponds to  collectionFormat: multi from OpenAPI 2.0.

I tried, with no luck (and not so much effort), to configure `collectionFormat: multi`, but I don't know if it's implemented in this library.
Any updates @axnsan12 ?

Thanks!

**Comment by marcorichetta** [comment]
Wow, almost a year after I got into the same rabbit hole jaja Luckily, this time I could find a workaround.

### Old way
```py
# serializers.py
class LegajosQueryParamsSerializer(serializers.Serializer):

    legajos = serializers.ListField(
        child=serializers.IntegerField(required=True),
        allow_empty=False
    )

# views.py
@swagger_auto_schema(
        query_serializer=LegajosQueryParamsSerializer
    )
    def get(self, request: Request):
        serializer = LegajosQueryParamsSerializer(data=request.query_params)
        serializer.is_valid(raise_exception=True)
```

This produces a **validation error**
<img src="https://user-images.githubusercontent.com/19599150/171261138-a52a702e-a641-456e-9f98-1717322976ed.png" alt="validation error" width="400"/>

### Workaround
Basically define the manual parameter. Obviously not scalable.
```py
#views.py
@swagger_auto_schema(
        manual_parameters=[openapi.Parameter(
            "legajos",
            openapi.IN_QUERY,
            "Legajo", 
            type=openapi.TYPE_ARRAY,
            items=openapi.Items(type=openapi.TYPE_INTEGER),
            required=True, 
            collection_format='multi' # This does the magic
        )],
)
```
<img src="https://user-images.githubusercontent.com/19599150/171261795-8d41993a-6a79-42e9-9824-fa9de710ee5a.png" alt="validation error" width="400"/>

**Comment by paaksing** [comment]
@marcorichetta Here is my SCALABLE workaround:
```python

from drf_yasg.generators import OpenAPISchemaGenerator
from drf_yasg import openapi


class ModifiedSchemaGenerator(OpenAPISchemaGenerator):

    def get_schema(self, request=None, public=False):
        schema = super().get_schema(request, public)
        schema.schemes = ["http", "https"]
        return schema

    def get_operation(self, *args, **kwargs):
        operation = super().get_operation(*args, **kwargs)
        for parameter in operation.parameters:
            if "type" in parameter and parameter["type"] == openapi.TYPE_ARRAY:
                parameter["collection_format"] = "multi"
                if hasattr(parameter["items"], "enum"):
                    parameter["description"] = "Use CTRL key to multi select"
        return operation

```

**Comment by kfke** [comment]
Hi all, it should be like this  "parameter["collectionFormat"] = "multi"" , collectionFormat not collection_format
